src/ast.rs

```diff
@@ -0,0 +1,92 @@
+pub struct Item {
+    pub is_pub: Boolean,
+    pub ident: String,
+    pub kind: ItemKind,
+}
+
+pub enum ItemKind {
+    Fn(Fn),
+    Struct,
+    Enum,
+    Mod,
+    Use,
+    Trait,
+    TyAlias,
+}
+
+pub struct Fn {
+    pub generics: Generics,
+    pub sig: FnSig,
+}
+
+pub struct Generics {
+    pub params: Vec<GenericParam>,
+    pub where_clause: Vec<GenericParam>,
```

Comment:
I think it would be better to represent `where_clause` in a similar form from rustc ast, which is
```rust
/// A single predicate in a where-clause.
#[derive(Clone, Encodable, Decodable, Debug)]
pub enum WherePredicate {
    /// A type bound (e.g., `for<'c> Foo: Send + Clone + 'c`).
    BoundPredicate(WhereBoundPredicate),
    /// A lifetime predicate (e.g., `'a: 'b + 'c`).
    RegionPredicate(WhereRegionPredicate),
    /// An equality predicate (unsupported).
    EqPredicate(WhereEqPredicate),
}
```

In rustc ast, `WhereClause` just contains everything within `WherePredicate`, except `has_where_token` which doesn't seem to be required for us.

Reply:
lifetime doesn't exist in ts, and not need to support unsupported, so I modified it with just one variant.
It make sense to maintain original structure as much as possible because need of more information is possible and also unnecessary preprocessing is possible, and original structure can role as standard of this.
Let's make AST bigger or smaller after some progress of parsing or transpiling.